File: Python/191. Number of 1 Bits.py
class Solution1(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        return bin(n).count('1')

class Solution2(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans = 0
        while n:
            n &= (n - 1)
            ans += 1
        return ans

File: C++/191. Number of 1 Bits.cpp
class Solution1 {
public:
    int hammingWeight(uint32_t n) {
        bitset<100> s(n);
        return s.count();
    }
};

class Solution2 {
public:
    int hammingWeight(uint32_t n) {
        int ans = 0;
        while(n)
        {
            n &= (n-1);
            ans++;
        }
        return ans;
    }
};